### ISSUE TITLE
g++5.4.1 fails to compile index-util

### ISSUE THREAD

**Issue by jsteemann** [iresearch-toolkit/iresearch#1]
When compiling everything with g++-5.4.1 (g++ (Ubuntu 5.4.1-2ubuntu1~16.04) 5.4.1 20160904
) the compiler reports the following errors:
```
/home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp: In function ‘int dump(const string&, std::ostream&)’:
/home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:34:3: error: reference to ‘irs’ is ambiguous
   irs::fs_directory dir(path);
   ^
In file included from /home/steemann/Downloads/iresearch-build/iresearch/core/index/directory_reader.hpp:15:0,
                 from /home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:12:
/home/steemann/Downloads/iresearch-build/iresearch/core/shared.hpp:164:28: note: candidates are: namespace irs = iresearch;
 namespace irs = ::iresearch;
                            ^
/home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:26:26: note:                 namespace <unnamed>::irs = iresearch;
 namespace irs = iresearch;
                          ^
/home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:35:17: error: reference to ‘irs’ is ambiguous
   auto reader = irs::directory_reader::open(dir, irs::formats::get("1_0"));
                 ^
In file included from /home/steemann/Downloads/iresearch-build/iresearch/core/index/directory_reader.hpp:15:0,
                 from /home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:12:
/home/steemann/Downloads/iresearch-build/iresearch/core/shared.hpp:164:28: note: candidates are: namespace irs = iresearch;
 namespace irs = ::iresearch;
                            ^
/home/steemann/Downloads/iresearch-build/iresearch/index-util/main.cpp:26:26: note:                 namespace <unnamed>::irs = iresearch;
 namespace irs = iresearch;
                          ^
```
I was able to work around that by applying the following patch locally:
```diff
diff --git a/index-util/main.cpp b/index-util/main.cpp
index 19e2e8e..b7b24d3 100644
--- a/index-util/main.cpp
+++ b/index-util/main.cpp
@@ -23,7 +23,6 @@ namespace {
 
 namespace po = boost::program_options;
 namespace fs = boost::filesystem;
-namespace irs = iresearch;
 
 }
 
@@ -31,6 +30,7 @@ static const std::string INDEX_DIR = "index-dir";
 static const std::string OUTPUT = "out";
 
 int dump(const std::string& path, std::ostream& stream) {
+  namespace irs = ::iresearch;
   irs::fs_directory dir(path);
   auto reader = irs::directory_reader::open(dir, irs::formats::get("1_0"));
 
```

**Comment by gnusi** [comment]
Fixed in 3053085ef0d9def6dcc4694cfbe238543395f3fb

**Comment by jsteemann** [comment]
Thanks, now it compiles without the error.
